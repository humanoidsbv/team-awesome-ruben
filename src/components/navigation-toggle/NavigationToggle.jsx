import React from 'react';
import PropTypes from 'prop-types';

import styles from './NavigationToggle.module.css';
import MenuIcon from '../../../assets/icons/icon-menu.svg';
import MenuClose from '../../../assets/icons/icon-close.svg';

const NavigationToggle = ({ isMenuVisible, toggleMenu }) => (
  <button
    className={`${styles.toggle} ${isMenuVisible ? styles.menuVisible : ''}`}
    onClick={toggleMenu}
    type="button"
  >
    <MenuIcon className={styles.menuIcon} />
    <MenuClose className={styles.closeIcon} />
  </button>
);

NavigationToggle.propTypes = {
  isMenuVisible: PropTypes.bool.isRequired,
  toggleMenu: PropTypes.func.isRequired
};

export default NavigationToggle;
