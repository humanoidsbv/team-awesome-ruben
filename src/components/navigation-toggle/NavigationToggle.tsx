import React from 'react';

import MenuClose from '../../assets/icons/icon-close.svg';
import MenuIcon from '../../assets/icons/icon-menu.svg';
import styles from './NavigationToggle.module.css';

interface NavigationToggle {
  isMenuVisible: boolean;
  toggleMenu: () => void;
}

const NavigationToggle = ({
  isMenuVisible,
  toggleMenu
}: NavigationToggle): React.ReactElement => (
  <button
    className={`${styles.toggle} ${isMenuVisible ? styles.menuVisible : ''}`}
    onClick={toggleMenu}
    type="button"
  >
    <MenuIcon className={styles.menuIcon} />
    <MenuClose className={styles.closeIcon} />
  </button>
);

export default NavigationToggle;
