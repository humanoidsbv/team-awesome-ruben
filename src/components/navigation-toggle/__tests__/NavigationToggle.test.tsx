import * as React from 'react';
import { shallow } from 'enzyme';

import NavigationToggle from '../NavigationToggle';

it('matches component output with expected output', () => {
  const navigationToggle = shallow(
    <NavigationToggle isMenuVisible={false} toggleMenu={() => {}} />
  );
  expect(navigationToggle).toMatchSnapshot();
});
