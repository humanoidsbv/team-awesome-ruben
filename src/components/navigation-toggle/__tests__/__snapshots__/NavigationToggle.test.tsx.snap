// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component output with expected output 1`] = `
<button
  className="toggle "
  onClick={[Function]}
  type="button"
>
  <test-file-stub
    className="menuIcon"
  />
  <test-file-stub
    className="closeIcon"
  />
</button>
`;
